app.js

```diff
@@ -0,0 +1,159 @@
+// Usage:
+// node app.js -r -c config.yaml -u "http://localhost:9000" # remember to add the registration!
+// node app.js -p 9000 -c config.yaml
+var qs = require('querystring');
+var requestLib = require("request");
+var bridge;
+
+function startServer(config, rooms, callback) {
+    var createServer;
+    if (config.tls) {
+        var fs = require("fs");
+        var tls_options = {
+            key: fs.readFileSync(config.tls.key_file),
+            cert: fs.readFileSync(config.tls.crt_file)
+        };
+        createServer = function(cb) { return require("https").createServer(tls_options, cb); };
+    } else {
+        createServer = require("http").createServer;
+    }
+
+    createServer(function(request, response) {
+        console.log(request.method + " " + request.url);
+
+        var body = "";
+        request.on("data", function(chunk) {
+            body += chunk;
+        });
+
+        request.on("end", function() {
+            var params = qs.parse(body);
+            if (params.user_id !== "USLACKBOT") {
+                var intent = bridge.getIntent("@" + config.username_prefix + params.user_name + ":" + config.homeserver.server_name);
+                if (rooms.knowsSlackChannel(params["channel_id"])) {
+                    var roomID = rooms.matrixRoomID(params["channel_id"]);
+                    if (roomID) {
+                        intent.sendText(roomID, params.text);
+                    } else {
+                        console.log(
+                            "Ignoring message for slack channel " +
+                            "with unknown matrix ID: " + params["channel_id"] +
+                            " (" + params["channel_name"] + ")"
+                        );
+                    }
+                }
+            }
+            response.writeHead(200, {"Content-Type": "application/json"});
+            response.write(JSON.stringify({}));
+            response.end();
+        });
+    }).listen(config.slack_hook_port, function() {
+        var protocol = "http";
+        if (config.tls) {
+            protocol = "https";
+        }
+        console.log("Slack-side listening on port " + config.slack_port + " over " + protocol);
+        callback();
+    });
+}
+
+var Cli = require("matrix-appservice-bridge").Cli;
+var Bridge = require("matrix-appservice-bridge").Bridge;
+var AppServiceRegistration = require("matrix-appservice").AppServiceRegistration;
```

Comment:
The newest HOW-TO code replaces the require from `"matrix-appservice"` to `"matrix-appservice-bridge"